src/plugins/collision/collider/constructor.rs

```diff
@@ -0,0 +1,788 @@
+                Some(ColliderConstructorHierarchyConfig::from_constructor(shape)),
+            );
+        }
+        self
+    }
+
+    /// Specifies the [`CollisionLayers`] used for an entity with the given `name`.
+    pub fn with_layers_for_name(self, name: &str, layers: CollisionLayers) -> Self {
+        self.with_config_for_name(name, |config| config.layers = Some(layers))
+    }
+
+    /// Specifies the [`ColliderDensity`] used for an entity with the given `name`.
+    pub fn with_density_for_name(self, name: &str, density: impl Into<ColliderDensity>) -> Self {
+        let density = density.into();
+        self.with_config_for_name(name, |config| config.density = Some(density))
+    }
+
+    /// Sets the [`ColliderConstructor`] for the entity associated with the given `name` to `None`,
+    /// skipping collider generation for it.
+    pub fn without_constructor_for_name(mut self, name: &str) -> Self {
+        self.config.insert(name.to_string(), None);
+        self
+    }
+
+    fn with_config_for_name(
+        mut self,
+        name: &str,
+        mut mutate_config: impl FnMut(&mut ColliderConstructorHierarchyConfig),
+    ) -> Self {
+        if let Some(Some(config)) = self.config.get_mut(name) {
+            mutate_config(config);
+        } else {
+            let mut config = self.base_constructor_hierarchy_data().unwrap_or_else(||
+                panic!("Failed to configure collider constructor for \"{name}\" because it has no associated constructor. \
+                Either specify a default constructor by passing one to `ColliderConstructorHierarchy::new` or call `with_constructor_for_name` first, \
+                or enable the `collider-from-mesh` feature to use a default construction method by reading the mesh."));
+            mutate_config(&mut config);
+            self.config.insert(name.to_string(), Some(config));
+        }
+        self
+    }
+
+    fn base_constructor_hierarchy_data(&self) -> Option<ColliderConstructorHierarchyConfig> {
+        self.default_constructor
+            .clone()
+            .map(ColliderConstructorHierarchyConfig::from_constructor)
+            .or({
+                #[cfg(all(feature = "3d", feature = "collider-from-mesh"))]
+                {
+                    Some(ColliderConstructorHierarchyConfig::default())
+                }
+                #[cfg(not(all(feature = "3d", feature = "collider-from-mesh")))]
+                {
+                    None
+                }
+            })
+    }
+}
+
+/// Configuration for a specific collider generated from a scene using [`ColliderConstructorHierarchy`].
+#[derive(Clone, Debug, PartialEq, Reflect)]
+#[reflect(Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serialize", reflect(Serialize, Deserialize))]
+#[cfg_attr(all(feature = "3d", feature = "collider-from-mesh"), reflect(Default))]
+pub struct ColliderConstructorHierarchyConfig {
```

Comment:
It's weird how everything but the constructor is optional. With this PR, wouldn't it make sense to want to override only the `layers` and leave the default constructor? I'd suggest making the constructor an `Option` as well.

This will also remove the special-cased `Default` and the need for `ColliderConstructorHierarchyConfig::from_constructor`

Reply:
Yeah, I agree. I made the constructor an `Option` and reworked the related logic a bit